src/Sarif.Viewer.VisualStudio/ErrorList/SarifAbsentResultsTableDataSource.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // A "stub" data source whose sole purpose is to cause the Category column
+    // to be displayed. It never actually adds entries to the Error List.
+    //
+    // For an explanation of why these this source is necessary, see the comment
+    // near the top of SarifResultTableEntry.cs
+    internal class SarifAbsentResultsTableDataSource : SarifTableDataSourceBase
+    {
+        private static SarifAbsentResultsTableDataSource _instance;
+
+        private SarifAbsentResultsTableDataSource()
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore VSTHRD108 // Assert thread affinity unconditionally
+
+                this.Initialize(SarifResultTableEntry.AbsentResultColumns);
+            }
+        }
+
+        public static SarifAbsentResultsTableDataSource Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new SarifAbsentResultsTableDataSource();
+                }
+
+                return _instance;
+            }
+        }
+
+        public override string Identifier => $"{Guids.GuidVSPackageString}-{nameof(SarifAbsentResultsTableDataSource)}";
+
+        public override string DisplayName => Resources.ErrorListAbsentResultsDataSourceDisplayName;
+
+        public override IDisposable Subscribe(ITableDataSink sink) => new StubDisposable();
```

Comment:
>Subscribe [](start = 36, length = 9)

Can you verify that calls to this method are sufficiently 'not chatty' to warrant creating a new object on each call, rather than, say, initializing and returning a class member?

Reply:
Yes: VS calls this once on each new table data source that is registered, and there are only 3 of them.

---
In reply to: [538753999](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538753999) [](ancestors = 538753999)

Reply:
Edit: Sorry, once per _sink_, on each new table data source. I admit I don't know how many sinks there are.

---
In reply to: [538775760](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538775760) [](ancestors = 538775760,538753999)